packages/venia-concept/src/components/MiniCart/editItem.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect } from 'react';
+import { bool, func, object } from 'prop-types';
+
+import { useQuery } from '@magento/peregrine';
+
+import { loadingIndicator } from 'src/components/LoadingIndicator';
+import PRODUCT_DETAILS from 'src/queries/getProductDetailByName.graphql';
+
+import CartOptions from './cartOptions';
+
+const EditItem = props => {
+    // Props.
+    const { endEditItem, isUpdatingItem, item, updateItemInCart } = props;
+
+    // We must have an item to edit.
+    if (!item) {
+        // TODO: log an error?
+        return null;
+    }
+
+    // State / Hooks.
+    const [queryResult, queryApi] = useQuery(PRODUCT_DETAILS);
+    const { data, error, loading } = queryResult;
+    const { runQuery, setLoading } = queryApi;
+
+    // Members.
+    const itemHasOptions = item.options.length > 0;
+
+    // Don't have to run the query for non-configurable items.
+    if (!itemHasOptions) {
+        return (
+            <CartOptions
+                cartItem={item}
+                configItem={{}}
+                endEditItem={endEditItem}
+                isUpdatingItem={isUpdatingItem}
+                updateCart={updateItemInCart}
+            />
+        );
+    }
+
+    // Run the query once on mount and again whenever the
+    // item being edited changes.
+    useEffect(() => {
+        setLoading(true);
+        runQuery({
+            variables: {
+                name: item.name,
+                onServer: false
+            }
+        });
+    }, [item]);
+
+    /*
+     * Render.
+     */
+
+    // We don't have data yet, we're either loading it or
+    // in an error situation.
+    if (!data) {
+        let status;
```

Comment:
Is there a default status we should use here when there's no data, but `error` and `loading` are false? If not, simplify this down `const status = error ? <span>...</span> : loadingIndicator` (assumes empty data and loading are the same state).